packages/core/src/shared/wizards/stateController.ts

```diff
@@ -105,6 +118,12 @@ export class StateMachineController<TState> {
         }
 
         if (isMachineResult(result)) {
+            if (result.controlSignal === ControlSignal.Skip) {
+                /**
+                 * Depending on current wizard direction, skip signal get converted to forward or backward control signal
+                 */
+                result.controlSignal = this.direction === DIRECTION.Forward ? undefined : ControlSignal.Back
```

Comment:
How would this work on Wizard prompter? Will it able to identify the direction of it's parent wizard?

Reply:
The `WizardPrompter` is still considered a prompter. It is not responsible for identifying parent wizard direction. 

This state management functionality belongs to the `stateController` object within the Wizard class. Each Wizard is responsible for prompting, receiving, and processing answers from prompters (including `WizardPrompter`). Each answer contains data and a `ControlSignal` (undefined indicates no direction change). The Wizard class's `stateController` tracks the current direction and uses the `ControlSignal` from the prompter to manage direction changes.

How do we restore the state of `WizardPrompter` when a back button is clicked?
Well, we do not restore the state of the `WizardPrompter` object, we always instantiate a new instance of this class. We, however, restore the state of Wizard class used for instantiate the `WizardPrompter`.  Again, the (parent) wizard class is responsible for this [(reference)](https://github.com/aws/aws-toolkit-vscode/blob/83cc0ff4656ce0130b41056c81dd5c9c8834a36a/packages/core/src/shared/wizards/wizard.ts#L108-L145)